test/control.js

```diff
@@ -79,3 +79,48 @@ metatests.test('sequential with error', test => {
     test.end();
   });
 });
+
+metatests.test('runIf run asyncFn', test => {
+  const condition = true;
+  const asyncFn = (arg1, arg2, cb) => {
+    process.nextTick(() => {
+      cb(null, { arg1, arg2 });
+    });
+  };
+
+  metasync.runIf(condition, asyncFn, 'val1', 'val2', (err, result) => {
```

Comment:
Could you wrap the callbacks in `test.mustCall` and `test.mustNotCall` where needed?